src/Controller/ActivityController.php

```diff
@@ -403,4 +404,53 @@ public function editAction(Activity $activity, ActivityRepository $activityRepos
 
         return new JsonResponse(['message' => 'Activity successfully edited!'], Response::HTTP_OK);
     }
+
+    /**
+     * Apply for an Activity.
+     * @Rest\Post("/{id}/apply")
+     * @SWG\Post(
+     *     tags={"Activity"},
+     *     summary="Apply for an Activity.",
+     *     description="Apply for an Activity.",
+     *     operationId="applyForActivity",
+     *     produces={"application/json"},
+     *     @SWG\Parameter(
+     *     description="ID of Activity to apply",
+     *     in="path",
+     *     name="id",
+     *     required=true,
+     *     type="integer",
+     * )
+     * )
+     * @SWG\Response(
+     *     response=401,
+     *     description="Unauthorized.",
+     *     @SWG\Schema(
+     *     @SWG\Property(property="code", type="integer", example=401),
+     *     @SWG\Property(property="message", type="string", example="JWT Token not found"),
+     *     )
+     * )
+     * @SWG\Response(
+     *     response="200",
+     *     description="Successfull operation!",
+     *     @SWG\Schema(
+     *     @SWG\Property(property="message", type="string", example="Applied with succes!"),
+     *     )
+     * )
+     * @SWG\Response(
+     *     response=404,
+     *     description="Activity not found.",
+     * )
+     * @param Activity $activity
+     * @param ActivityUserRepository $repo
+     * @return JsonResponse
+     * @throws ORMException
+     * @throws OptimisticLockException
+     */
+    public function applyForActivity(Activity $activity, ActivityUserRepository $repo): JsonResponse
+    {
+        $applierUser = $this->getUser();
+        $repo->apply($activity, $applierUser);
```

Comment:
Before `apply()` check if user has already applied. He can't apply twice to the same activity.

Reply:
Also check status and deadline.